File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        stack=[]
        for b in s:
            if stack and stack[-1]=="(" and b==")":
                stack.pop()
            elif stack and stack[-1]=="{" and b=="}":
                stack.pop()
            elif stack and stack[-1]=="[" and b=="]":
                stack.pop()
            else:
                stack.append(b)
        return len(stack)==0

File: Data Structures & Algorithms/daily-temperatures/submission-1.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        let res=Array(temperatures.length).fill(0)
        let stack=[]
        for(let i=0;i<temperatures.length;i++){
            const t=temperatures[i]
            
            while(stack.length>0 && t>stack[stack.length-1][1]){
                const [idx,temp]=stack.pop()
                res[idx]=i-idx
            }
            stack.push([i,t])

        }
        return res
    }
}
